### DISCUSSION TITLE
what exactly are 'non-positional formatting attributes'?

### DISCUSSION THREAD

**Discussion by lemzwerg** [w3c-cg/musicxml#517] [Q&A]
In the documentation of the `<direction>` element you can find the following.

> For a series of <direction-type> children, non-positional formatting attributes are carried over from the previous element by default.

I have difficulties to understand this sentence, since 'non-positional formatting attributes' is too vague to me.  What exactly are they?  Obviously, `default-[xy]` and `relative-[xy]` are covered.  But what about `[hv]align` and `justify`?

And for some attributes I have difficulties to believe that the cited sentence is actually valid.  For example, both `pedal` and `octave-shift` have a `type` attribute but with different allowed values.

**Comment by mscuthbert** [comment]
I'm going to wager a guess that no one has systematically defined what values are inherited and which aren't.  So let's figure out together a list of things that it makes most sense to include in the inheritance and which to exclude while maintaining the general principle of the current rule.

We might want to define it via %attribute-groups -- for instance position is not inherited:

```
 <xs:attributeGroup name="position">
        <xs:attribute name="default-x" type="tenths"/>
        <xs:attribute name="default-y" type="tenths"/>
        <xs:attribute name="relative-x" type="tenths"/>
        <xs:attribute name="relative-y" type="tenths"/>
    </xs:attributeGroup>
```

We might wish to define `type` as a special case where it is inherited if the object-type is the same and not if it is different.  Though that gets a bit close to backward-incompatible change of behavior rather than the type of documentation clarification that I think is best to focus on.

**Reply by lemzwerg** [reply]
Indeed, it would be ideal if pages like https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/wedge contained a fifth column (besides 'Name', 'Type', 'Required?', and 'Description') – say, 'Carried over?' – that gets automatically generated, and which contains this information.

However, for the moment I would be fully content to simply know exactly what gets carried over and what not. :slightly_smiling_face:

**Reply by lemzwerg** [reply]
Besides this information I think it would be great to have an example in the documentation that demonstrates how this works.
